### ISSUE TITLE
Can't complete out-of-stock orders in "Address" state

### ISSUE THREAD

**Issue by filipefurtad0** [openfoodfoundation/openfoodnetwork#8178]
## Description

Closely related to #8177.
An order which as gone through the the steps described in #8177 and which is in the `expected` Address state cannot be completed:
- The admin does not "see" any items on the item list
- Adding a different line item will trigger an out-of-stock message concerning the initially added (but out-of-stock) item
- The only way to complete this order is for the customer to update their own cart, and remove the aforementioned line item

## Expected Behavior

- The admin should be able to edit the order, and add an alternative item to the order (within stock restrictions).
- The Admin should be able to remove the initial out of stock item and this should then be mirrored on the cart on the shopper side (this should prevent the out-of-stock message displayed to the admin and concerning the initially added out-of-stock item)

## Actual Behaviour

The admin cannot edit the order (within stock restrictions).
Adding a different line item will trigger an out-of-stock message concerning the initially added (but out-of-stock) item.

## Steps to Reproduce

1. Follow the steps to reproduce on #8177

As an admin
2. Edit the order and add an additional item to the order

As a customer
3. Visit your cart; Notice the newly added line item, but also the warning concerning the line item out of stock.

As a an admin
4. Click on the Payments tab -> an error message concerning the initially (but out-of-stock) added item will be displayed

This order is only unblocked if the customer removes the out-of-stock item from the cart (using the trash-bin icon; the Update Quantities button is missing, this can be addressed in a separate issue).

## Animated Gif/Screenshot

![Peek 2021-09-13 10-50](https://user-images.githubusercontent.com/49817236/133064101-a22033db-ac8f-407b-8e3f-a3c5aab20fde.gif)

## Workaround

Admin -> No workaround
Customer -> Empty your cart and start over

## Severity

bug-s3: a feature is broken but there is a workaround

## Your Environment

* Version used: v4.1.1
* Browser name and version: Firefox 91
* Operating System and version (desktop or mobile): Ubuntu 20.04

**Comment by jaycmb** [comment]
On expected behaviour, should we add that the Admin can remove the initial out of stock item and this should then be mirrored on the cart on the shopper side?

As far as I understood the above described behaviour the Admin would be able to add items, but it would still be mandatory **for the shopper** to remove the out of stock item.

Or do we need to keep the 0 on the shopper side so that the user knows their item was out of stock?

**Comment by filipefurtad0** [comment]
> On expected behaviour, should we add that the Admin can remove the initial out of stock item and this should then be mirrored on the cart on the shopper side?

Agreed and updated :+1: 

It seems that line item which is out of stock (seen in the DB and associated with this order, quantity = 5) does not appear in the UI for the admin and only for the customer. Only an added line item (with stock, quantity = 2) appears for both:
![image](https://user-images.githubusercontent.com/49817236/133247217-fda12c07-bc98-44a7-862e-f43375b3239a.png)
